src/simple.jl

```diff
@@ -45,10 +45,11 @@ function kinsol(f, y0::Vector{Float64}, userdata::Any = nothing)
     #    where `y` is the input vector, and `fy` is the result of the function
     # y0, Vector of initial values
     # return: the solution vector
-    kmem = Handle(KINCreate())
-    if kmem == C_NULL
+    mem_ptr = KINCreate()
+    if mem_ptr == C_NULL
```

Comment:
A matter of taste (or Sundials.jl code style guidelines), but I prefer `(mem_ptr == C_NULL) && error("...")`